Testnet/NonFungibleToken/NonFungibleToken/NonFungibleToken.cs

```diff
@@ -227,10 +238,11 @@ public void TransferFrom(Address from, Address to, UInt256 tokenId)
     /// <param name="tokenId">ID of the token to be approved.</param>
     public void Approve(Address approved, UInt256 tokenId)
     {
-        CanOperate(tokenId);
-        ValidateToken(tokenId);
-
         Address tokenOwner = GetIdToOwner(tokenId);
+        CanOperate(tokenOwner);
+
+        EnsureAddressIsNotEmpty(tokenOwner);
```

Comment:
Same comment here regarding the address not empty check, CanOperate would fail first, think you can remove the extra empty address check here.